build-tools/cmd/zz-e2e-perf.go

```diff
@@ -0,0 +1,460 @@
+	buildCmd.Flags().StringVar(&obj.flags.IgnoreFile, "ignore-file", ".gitignore", "Name of the file with files to exclude (in the format of .gitignore)")
+
+	// "push" sub-command
+	pushCmd := &cobra.Command{
+		Use:   "push",
+		Short: "push Docker image",
+		Long: fmt.Sprintf(`Pushes the pre-built Docker image for a %s test app.
+The image must have been aleady built using the "build" sub-command.
+`, cmdType),
+		RunE: obj.pushCmd,
+	}
+	pushCmd.Flags().StringVarP(&obj.flags.Name, "name", "n", "", "Name of the app")
+	pushCmd.MarkFlagRequired("name")
+	pushCmd.Flags().StringVar(&obj.flags.DestRegistry, "dest-registry", "", "Registry for the Docker image")
+	pushCmd.MarkFlagRequired("dest-registry")
+	pushCmd.Flags().StringVar(&obj.flags.DestTag, "dest-tag", "", "Tag to apply to the Docker image")
+	pushCmd.MarkFlagRequired("dest-tag")
+
+	// "build-and-push" sub-command
+	buildAndPushCmd := &cobra.Command{
+		Use:   "build-and-push",
+		Short: "Build and push Docker image",
+		Long: fmt.Sprintf(`Single command to build and push the Docker image for a %s test app.
+If the "--cahce-registry" option is set and the image exists in the cache, it will be copied directly from the cache, without pulling it locally first.
+`, cmdType),
+		RunE: obj.buildAndPushCmd,
+	}
+	buildAndPushCmd.Flags().StringVarP(&obj.flags.Name, "name", "n", "", "Name of the app")
+	buildAndPushCmd.MarkFlagRequired("name")
+	buildAndPushCmd.Flags().StringVarP(&obj.flags.AppDir, "appdir", "d", "", "Directory where the test apps are stored")
+	buildAndPushCmd.MarkFlagRequired("appdir")
+	buildAndPushCmd.MarkFlagDirname("appdir")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.DestRegistry, "dest-registry", "", "Registry for the Docker image")
+	buildAndPushCmd.MarkFlagRequired("dest-registry")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.DestTag, "dest-tag", "", "Tag to apply to the Docker image")
+	buildAndPushCmd.MarkFlagRequired("dest-tag")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.CacheRegistry, "cache-registry", "", "Cache container registry (optional)")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.Dockerfile, "dockerfile", "Dockerfile", "Dockerfile to use")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.TargetOS, "target-os", runtime.GOOS, "Target OS")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.TargetArch, "target-arch", runtime.GOARCH, "Target architecture")
+	buildAndPushCmd.Flags().StringVar(&obj.flags.IgnoreFile, "ignore-file", ".gitignore", "Name of the file with files to exclude (in the format of .gitignore)")
+
+	// Register the commands
+	cmd.AddCommand(buildCmd)
+	cmd.AddCommand(pushCmd)
+	cmd.AddCommand(buildAndPushCmd)
+
+	return cmd
+}
+
+// Handler for the "build" sub-command
+func (c *cmdE2EPerf) buildCmd(cmd *cobra.Command, args []string) error {
+	// Target image and tag
+	destImage := c.getDestImage()
+
+	// If cache is enable, try pulling from cache first
+	cachedImage, err := c.getCachedImage()
```

Comment:
It took a bit for me to realize that the hash validation happens because the filepath is no longer there. Think you can add a comment here (or at the method) stating that it handles hash validation there?

Reply:
Done